socorro/submitter/breakpad_submitter_utilities.py

```diff
@@ -1,68 +1,58 @@
-from socorro.external.crashstorage_base import CrashStorageBase
 
 import os
 import urllib2
+
+from configman import Namespace
 import poster
+
+from socorro.external.crashstorage_base import CrashStorageBase
+
+
 poster.streaminghttp.register_openers()
 
 
 class BreakpadPOSTDestination(CrashStorageBase):
-    """This a crashstorage derivative that pushes a crash out to a
-    Socorro collector waiting at a url.
+    """This a crashstorage derivative that pushes a crash out to a Socorro collector waiting at a url.
 
     """
     required_config = Namespace()
     required_config.add_option(
-        'url',
-        short_form='u',
-        doc="The url of the Socorro collector to submit to",
-        default="http://127.0.0.1:8882/submit"
+        'urls',
+        doc='One or more urls to submit to separated by commas',
+        default="http://127.0.0.1:8888/submit"
     )
-    required_config.add_option(
-        'echo_response',
-        short_form='e',
-        doc="echo the submission response to stdout",
-        default=False
-    )
-
-    def __init__(self, config, quit_check_callback=None):
-        super(BreakpadPOSTDestination, self).__init__(
-            config,
-            quit_check_callback
-        )
-        self.hang_id_cache = dict()
 
     def save_raw_crash_with_file_dumps(self, raw_crash, dumps, crash_id):
+        """Saves a raw crash by wrapping it up in a multipart/form-data payload and submitting it as an HTTP
+        POST to a collector
+
+        """
+        urls = self.config.urls.split(',')
+
         try:
+            # Create raw crash with file pointers to minidumps for poster to pull from
             for dump_name, dump_pathname in dumps.iteritems():
                 if not dump_name:
                     dump_name = self.config.source.dump_field
                 raw_crash[dump_name] = open(dump_pathname, 'rb')
+
+            # Get uuid for logging purposes
+            uuid = raw_crash.get('uuid', 'NO UUID')
+
+            # Build the payload
             datagen, headers = poster.encode.multipart_encode(raw_crash)
-            request = urllib2.Request(
-                self.config.url,
-                datagen,
-                headers
-            )
-            submission_response = urllib2.urlopen(request).read().strip()
-            try:
+
+            # Submit payload to all urls
+            for url in urls:
+                request = urllib2.Request(url, datagen, headers)
+                submission_response = urllib2.urlopen(request).read().strip()
                 self.config.logger.debug(
-                    'submitted %s (original crash_id)',
-                    raw_crash['uuid']
+                    'submitted %s to %s; response %s', uuid, url, submission_response
                 )
-            except KeyError:
-                pass
-            self.config.logger.debug(
-                'submission response: %s',
-                submission_response
-            )
```

Comment:
I collapsed all this logging into a single line. Since we're using `uuid` which handles the "no uuid" case, we don't need to catch `KeyError`.